tests/functional/ingestion/IngestionReader.js

```diff
@@ -0,0 +1,247 @@
+        assert.deepStrictEqual(entry, this.expectedEntry);
+        this.publish(this.config.topic, `${entry.bucket}/${entry.key}`,
+            JSON.stringify(entry));
+    }
+}
+
+const expectedEntry = {
+    type: 'put',
+    bucket: 'zenkobucket',
+    key: 'object1',
+    value: '{"owner-display-name":"test_1518720219","owner-id":' +
+    '"94224c921648ada653f584f3caf42654ccf3f1cbd2e569a24e88eb460f2f84d8",' +
+    '"content-length":0,"content-md5":"d41d8cd98f00b204e9800998ecf8427e",' +
+    '"x-amz-version-id":"null","x-amz-server-version-id":"",' +
+    '"x-amz-storage-class":"STANDARD","x-amz-server-side-encryption":"",' +
+    '"x-amz-server-side-encryption-aws-kms-key-id":"",' +
+    '"x-amz-server-side-encryption-customer-algorithm":"",' +
+    '"x-amz-website-redirect-location":"","acl":{"Canned":"private",' +
+    '"FULL_CONTROL":[],"WRITE_ACP":[],"READ":[],"READ_ACP":[]},"key":"",' +
+    '"location":[],"isDeleteMarker":false,"tags":{},"replicationInfo":' +
+    '{"status":"","backends":[],"content":[],"destination":"","storageClass":' +
+    '"","role":"","storageType":"","dataStoreVersionId":""},"dataStoreName":' +
+    '"us-east-1","last-modified":"2018-02-16T22:43:37.174Z",' +
+    '"md-model-version":3}',
+};
+
+const extIngestionQP = new TestIngestionQP({
+    config: testConfig.extensions.ingestion,
+    expectedEntry });
+
+const zkClient = zookeeper.createClient('localhost:2181', {
+    autoCreateNamespace: true,
+});
+
+describe('ingestion reader tests with mock', () => {
+    let httpServer;
+    let batchState;
+
+    before(done => {
+        const mongoUrl =
+            `mongodb://${testConfig.queuePopulator.mongo.replicaSetHosts}` +
+            '/db?replicaSet=rs0';
+        async.waterfall([
+            next => {
+                MongoClient.connect(mongoUrl, {}, (err, client) => {
+                    if (err) {
+                        next(err);
+                    }
+                    this.client = client;
+                    this.db = this.client.db('metadata', {
+                        ignoreUndefined: true,
+                    });
+                    next();
+                });
+            },
+            next => this.db.createCollection('PENSIEVE', err => {
+                assert.ifError(err);
+                return next();
+            }),
+            next => {
+                this.m = this.db.collection('PENSIEVE');
+                this.m.insertOne(dummyPensieveCredentials, {});
+                return next();
+            },
+            next => {
+                this.m.insertOne({
+                    _id: 'configuration/overlay-version',
+                    value: 6,
+                }, {});
+                return next();
+            },
+            next => {
+                this.m.insertOne(dummySSHKey, {});
+                return next();
+            },
+            next => {
+                zkClient.connect();
+                zkClient.once('error', next);
```

Comment:
This event can be caused multiple times, try protecting it with `cbOnce` utility in Backbeat.